### DISCUSSION TITLE
Error: Could not find a relationship between 'posts' and 'posts' in the schema cache

### DISCUSSION THREAD

**Discussion by rahuldubey11** [supabase/supabase#10527] [Questions] [answered]
Hello all,
When I run this query:

```
final response = await supabase
.from('posts')
.select("*, repostsCount:posts!posts_repostID_fkey(count)")
.execute(count: CountOption.exact);
```

I get this error:

> response: {data: null, status: 400, error: {message: Could not find a relationship between 'posts' and 'posts' in the schema cache, code: PGRST200, details: null, hint: Verify that 'posts' and 'posts' exist in the schema 'public' and that there is a foreign key relationship between them. If a new relationship was created, try reloading the schema cache.}, count: null}

I can confirm that the foreign key is setup correctly. What's strange is that the exact same query was working fine for months till last week and suddenly I started getting this error.

As suggested in the error description, I have tried reloading the schema cache using:
`NOTIFY pgrst, 'reload schema'`

But it doesn't help. I have been stuck here for days and couldn't find a solution so, finally I am asking for help.
Thank you!

**Comment by steve-chavez** [comment, marked as answer]
Hello,

Could you send us your project ref to support@supabase.io?

**Reply by rahuldubey11** [reply]
Hello, 

Project ref sent via email.

**Reply by steve-chavez** [reply]
Ok, I do see that posts has a recursive relationship to itself.

Can you try using the column name instead of the FK name for the hint? Like:

```js
final response = await supabase
.from('posts')
.select("*, repostsCount:posts!repostID(count)")
.execute(count: CountOption.exact);
```

**Reply by rahuldubey11** [reply]
That was it! It's working now. Thank you so much!!!

**Reply by skamensky** [reply]
This fixed the error for me, but it guessed the relationship incorrectly. It related many objects to one object as an array, and the rest of the objects had an empty array. What should have happened is that each object should have a single object.

 I would think foreign key has the most granular level of specificity, not the column name...

**Reply by sameernshamsudheen** [reply]
const { data, error: deleteError } = await supabase
      .from("saved_jobs")
      .delete()
      .eq("job_id", 1);   this is my query  iam using supabase  
![supbase](https://github.com/user-attachments/assets/f8bf79c6-d09c-4edf-8274-751b8594da11)
iam  giving  user_id and job _id  but  cannot  delete  the  database data  how can i do it  can anyone help me

**Reply by sameernshamsudheen** [reply]
{
    "code": "PGRST204",
    "details": null,
    "hint": null,
    "message": "Could not find the '0' column of 'saved_jobs' in the schema cache"
}   this is the error iam getting

**Reply by steve-chavez** [reply]
Paste the full JS snippet, looks like you're adding an invalid column somewhere.

**Reply by sameernshamsudheen** [reply]
thank you steve  i have  resolved the issue

**Reply by HelcioAnicio** [reply]
> thank you steve i have resolved the issue
How did you resolve the issue? I'm with that issue
